src/components/frame/v5/pages/AgreementsPage/hooks.ts

```diff
@@ -15,23 +15,61 @@ import { notNull } from '~utils/arrays/index.ts';
 
 import { useFiltersContext } from './FiltersContext/FiltersContext.ts';
 
-export const useGetAgreements = () => {
+const QUERY_PAGE_SIZE = 20;
+
+const useGetAllAgreements = () => {
   const {
     colony: { colonyAddress },
   } = useColonyContext();
-  const { activeFilters, searchFilter } = useFiltersContext();
-  const { data, loading } = useGetColonyActionsQuery({
+
+  const { data, loading, fetchMore } = useGetColonyActionsQuery({
     variables: {
       colonyAddress,
       filter: {
         type: { eq: ColonyActionType.CreateDecisionMotion },
       },
       sortDirection: ModelSortDirection.Desc,
+      limit: QUERY_PAGE_SIZE,
     },
     fetchPolicy: 'network-only',
+    onCompleted: (newData) => {
+      if (newData?.getActionsByColony?.nextToken) {
+        fetchMore({
+          variables: { nextToken: newData?.getActionsByColony?.nextToken },
```

Comment:
I thought I'd just bring this up as well. And I'm not saying to do this in your PR. If the intention is to come up with a sort of infinite scrolling mechanism, we could either create our own hook that detects the scroll position of a container or use some library that can detect when a container reaches its bottom scroll threshold before firing the `fetchMore()` function.

And I say this because this `onCompleted` logic is set up to keep on calling the `GetColonyActionsQuery` until it fetches all the data, all in one go, which sort of defeats the purpose of pagination. You can prove this by setting `QUERY_PAGE_SIZE = 1` then open the GraphQL Network tab and filter the query by:
```
type":{"eq":"CREATE_DECISION_MOTION"}}
```
![Screenshot 2024-08-02 at 12 53 43](https://github.com/user-attachments/assets/a45574c3-db6a-4012-b27b-4c5e43ece17d)

You'll find that it will continue to query until all the data has been fetched, when `nextToken` becomes `null` essentially.

Our logic should be something like (this is future stuff, not in this PR):

```ts
<InfiniteScrollView onBottomReached={fetchMore}>
  {agreementItems.map(agreementItem => <AgreementItem agreementItem={agreementItem} />
</InfiniteScrollView>
```

Reply:
I half agree with this. Two things I want to point out:
- this is something for the far future, we have more pressing things to fix
- this needs to be a decision by product ultimately

But all in all I like your suggestion and support you coming up with more stuff like this in the future :tada: